jmix-security/security-flowui/src/main/java/io/jmix/securityflowui/view/resetpassword/ResetPasswordView.java

```diff
@@ -16,55 +16,227 @@
+        return userPasswordValue.getUsername();
+    }
+
+    @Supply(to = "passwordsDataGrid.username", subject = "renderer")
+    protected Renderer<UserPasswordValue> passwordsDataGridUsernameRenderer() {
+        return new TextRenderer<>(UserPasswordValue::getUsername);
+    }
+
+    @Supply(to = "passwordsDataGrid.password", subject = "renderer")
+    protected Renderer<UserPasswordValue> passwordsDataGridPasswordRenderer() {
+        return new ComponentRenderer<>(this::layoutFactory, this::passwordFieldInitializer);
+    }
+
+    protected void initExportAction() {
+        if (uiExportHelper == null) {
+            return;
+        }
+
+        exportActionButton = uiComponents.create(JmixButton.class);
+        exportActionButton.setText(messageBundle.getMessage("resetPasswordView.exportActionButton.text"));
+        exportActionButton.setIcon(VaadinIcon.FILE_TABLE.create());
+        exportActionButton.setVisible(false);
+        exportActionButton.addClickListener(this::onExportBtnClick);
+
+        buttonPanel.addComponentAtIndex(buttonPanel.indexOf(closeBtn), exportActionButton);
+    }
+
+    protected void onExportBtnClick(ClickEvent<Button> event) {
+        uiExportHelper.exportDataGrid(passwordsDataGrid);
+    }
+
+    @Override
+    public String getPageTitle() {
+        return isSingleSelected()
+                ? messageBundle.formatMessage("resetPasswordView.resetSinglePasswordTitle",
+                users.iterator().next().getUsername())
+                : messageBundle.getMessage("resetPasswordView.resetPasswordsTitle");
+    }
+
     @SuppressWarnings("unchecked")
-    @Subscribe("saveAction")
-    public void onSaveActionPerformed(ActionPerformedEvent event) {
-        Map<UserDetails, String> newPasswords = userManager.resetPasswords((Set<UserDetails>) users);
-        StringBuilder builder = new StringBuilder();
-        for (Map.Entry<UserDetails, String> entry : newPasswords.entrySet()) {
-            builder.append(entry.getKey().getUsername())
-                    .append("\t")
-                    .append(entry.getValue())
-                    .append("\n");
+    @Install(to = "userPasswordValueDl", target = Target.DATA_LOADER)
+    protected List<UserPasswordValue> userPasswordValuesDlLoadDelegate(LoadContext<UserPasswordValue> loadContext) {
+        if (!passwordsDataGrid.isVisible()) {
+            passwordsDataGrid.setVisible(true);
+
+            if (exportActionButton != null) {
+                exportActionButton.setVisible(true);
+            }
         }
-        readyPassword.setValue(builder.toString());
-        readyPassword.setVisible(true);
+
+        return userManager.resetPasswords((Set<UserDetails>) users)
+                .entrySet()
+                .stream()
+                .map(this::userPasswordValueMapper)
+                .toList();
+    }
+
+    @Subscribe("generateBtn")
+    protected void onGenerateBtnClick(ClickEvent<JmixButton> event) {
+        userPasswordValueDl.load();
```

Comment:
If there are a lot of users, UI freezes. Let's add progressbar and cancelable bg task